### ISSUE TITLE
Remove from TestProject any analysis/log messages on test results, this is done in depth in GatherTestInformation

### ISSUE THREAD

**Issue by fermadeiral** [Spirals-Team/repairnator#633]
```
01:26:15.070 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - Results :
01:26:15.070 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - 
01:26:15.070 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - Failed tests: 
01:26:15.070 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler -   LookupSnapshotTakerTest.testIOExceptionDuringLookupPersist Expected test to throw (an instance of io.druid.java.util.common.ISE and exception with message a string containing "Exception during serialization of lookups")
01:26:15.070 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler -   LocalDataSegmentPusherTest.testPushCannotCreateDirectory Expected test to throw (an instance of java.io.IOException and exception with message a string containing "Unable to create directory")
01:26:15.070 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler -   SegmentLoaderLocalCacheManagerTest.testRetryAllFail:307
01:26:15.070 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler -   SegmentLoaderLocalCacheManagerTest.testRetryAllFail:307
01:26:15.071 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler -   SegmentLoaderLocalCacheManagerTest.testRetrySuccessAtSecondLocation:252
01:26:15.071 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler -   SegmentLoaderLocalCacheManagerTest.testRetrySuccessAtSecondLocation:252
01:26:15.071 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - Tests in error: 
01:26:15.071 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler -   SQLMetadataRuleManagerTest.cleanup:204->dropTable:209 » CallbackFailed org.ski...
01:26:15.071 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler -   LookupReferencesManagerTest.testRealModeWithMainThread:472 »  test timed out a...
01:26:15.071 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - 
01:26:15.071 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - Tests run: 1220, Failures: 6, Errors: 2, Skipped: 18
01:26:15.071 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - 
```

[...]

`01:26:16.634 [main] DEBUG f.i.s.r.process.step.TestProject - 12 tests failed, 4 tests have error.`

It is probably being doubled.

**Comment by fermadeiral** [comment]
@surli, could you please explain to me a thing on `MavenFilterTestOutputHandler`?
In the test `TestMavenFilterTestOutputHandler.testFilterLogWillWork2()`, why you multiply by 2 the number of running tests and skipping tests?

```
assertEquals(77351*2, filterTest.getRunningTests());
assertEquals(0, filterTest.getFailingTests());
assertEquals(0, filterTest.getErroringTests());
assertEquals((52+6+1)*2, filterTest.getSkippingTests());
```

**Comment by surli** [comment]
If I remember well the purpose of `MavenFilterTestOutputHandler` is primarily to detect a test failure from the log. So I'm catching both intermediate results and sum-up results from the logs. Then in:

```
05:15:11.926 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - Running io.druid.concurrent.ExecsTest
05:15:13.454 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.526 sec - in io.druid.concurrent.ExecsTest
05:15:13.807 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler -
05:15:13.808 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - Results :
05:15:13.808 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler -
05:15:13.808 [Thread-8] DEBUG f.i.s.r.p.m.o.MavenFilterTestOutputHandler - Tests run: 324, Failures: 0, Errors: 0, Skipped: 1
```

I will get both the info about the 3 running test and the final sum-up about 324 tests: in the end all my results are multiplied by 2. 

On the global vision I did not really care at this step to get the right exact numbers: I mostly want to know if a test was failing or not, the right results should come from the surefire xml files.

**Comment by fermadeiral** [comment]
> I'm catching both intermediate results and sum-up results from the logs [...] in the end all my results are multiplied by 2.

I imagined that, but it's strange because in some projects that I test the final results are not multiplied by 2, I don't understand why, but anyway...

> On the global vision I did not really care at this step to get the right exact numbers: I mostly want to know if a test was failing or not

If the numbers are really not important for this step, the `MVN_GOAL_FAILED_WITH_TEST_FAILURE = "There are test failures"` by itself does the job you want.
My point is that I don't really think nice a log message with wrong info, so I see three options for me:
- To remove the computation of numbers from `MavenFilterTestOutputHandler`, just keep the thing I just mentioned;
- Or to remove the log messages that show wrong info;
- Or to skip the log messages when it's running for Bears, so nothing would affect for you.

WDYT?

> the right results should come from the surefire xml files

This is what I think too.

**Comment by fermadeiral** [comment]
In fact, I'm thinking at putting that log message I'm discussing here in the step `GatherTestInformation` instead of keeping it in `TestProject`. There is no such log message in `GatherTestInformation`, and there the numbers would be precise, and I also think it makes sense this info be logged from there.

**Comment by surli** [comment]
Actually I think we should get rid of `MavenFilterTestOutputHandler` and to rely only on `GatherTestInformation` by making `TestProject` a not blocking step. WDYT?

**Comment by fermadeiral** [comment]
> Actually I think we should get rid of MavenFilterTestOutputHandler and to rely only on GatherTestInformation by making TestProject a not blocking step. WDYT?

Hum, ok, let me think. In fact, let me check a think first with you before I agree with you. I'm checking the entire code of `TestProject`, and there is a piece that makes me believe that such step should be a blocking one; it's this piece:

```
int result = MavenHelper.MAVEN_ERROR;
        try {
            result = helper.run();
        } catch (InterruptedException e) {
            this.addStepError("Error while executing maven goal", e);
        }
```
Because if the step fails because of this part of the code, I think the pipeline should be stopped in `TestProject`. Please, tell me what you think about this piece of code, maybe I'm thinking wrong.

BUT, for the rest of the code in `TestProject`, I already agree with you: `TestProject` doesn't need to stop the pipeline, because the results of the tests would be analyzed in `GatherTestInformation`.

**Comment by fermadeiral** [comment]
In simple words, what I'm thinking is: `TestProject` is to execute the tests, so if something wrong happens during the execution of the tests, the pipeline should stop; but the pipeline should not stop depending on the test results, this is the job of `GatherTestInformation`. Do you agree with that?

**Comment by fermadeiral** [comment]
In my vision now, I think we could replace this:

```
        int result = MavenHelper.MAVEN_ERROR;
        try {
            result = helper.run();
        } catch (InterruptedException e) {
            this.addStepError("Error while executing maven goal", e);
        }

        if (result == MavenHelper.MAVEN_SUCCESS) {
            if (outputTestFilter.getRunningTests() > 0) {
                this.getLogger().debug(outputTestFilter.getRunningTests() + " tests has been launched but none failed.");
            } else {
                this.addStepError("No test recorded.");
            }
            return StepStatus.buildSuccess(this);
        } else {
            if (outputTestFilter.isFailingWithTest()) {
                this.getLogger().debug(outputTestFilter.getFailingTests() + " tests failed, "
                        + outputTestFilter.getErroringTests() + " tests have error.");
                return StepStatus.buildSuccess(this);
            } else {
                this.addStepError("Error while testing the project.");
                return StepStatus.buildError(this, PipelineState.NOTTESTABLE);
            }
        }
```
by only this:

```
        try {
            helper.run();
        } catch (InterruptedException e) {
            this.addStepError("Error while executing maven goal", e);
            return StepStatus.buildError(this, PipelineState.NOTTESTABLE);
        }
        return StepStatus.buildSuccess(this);
```

And then to log numbers in `GatherTestInformation`.

WDYT?

**Comment by surli** [comment]
ok with that.

**Comment by fermadeiral** [comment]
Deal then, I hope this works.

I will not touch in `MavenFilterTestOutputHandler` by now, but I will open another issue for a review of this class.

**Comment by surli** [comment]
fixed in #656
